src/Parser/CameraCalibrationParser.js

```diff
@@ -0,0 +1,80 @@
+import * as THREE from 'three';
+import OrientedImageCamera from '../Renderer/OrientedImageCamera';
+
+/**
+ * The CameraCalibrationParser module provide a [parse]{@link module:CameraCalibrationParser.parse}
+ * method that takes a JSON array of camera calibrations in and yields an array of THREE.Camera
+ *
+ * @module CameraCalibrationParser
+ */
+
+THREE.Matrix4.prototype.setMatrix3 = function setMatrix3(m) {
+    this.elements[0] = m.elements[0];
+    this.elements[1] = m.elements[1];
+    this.elements[2] = m.elements[2];
+    this.elements[4] = m.elements[3];
+    this.elements[5] = m.elements[4];
+    this.elements[6] = m.elements[5];
+    this.elements[8] = m.elements[6];
+    this.elements[9] = m.elements[7];
+    this.elements[10] = m.elements[8];
+    return this;
+};
+
+// the json format encodes the following transformation:
+// extrinsics: p_local = rotation * (p_world - position)
+// intrinsics: p_pixel = projection * p_local
+// distortion: p_raw = distortion(p_pixel)
+function parseCalibration(calibration, options) {
+    // parse intrinsics
+    const proj = calibration.projection;
+    const size = new THREE.Vector2().fromArray(calibration.size);
+    const focal = new THREE.Vector2(proj[0], proj[4]);
+    const center = new THREE.Vector2(proj[2], proj[5]);
+    const skew = proj[1];
+    var camera = new OrientedImageCamera(size, focal, center, options.near, options.far, skew);
+
+    // parse extrinsics: Object3d.matrix is from local to world
+    // p_world = position + transpose(rotation) * p_local
+    camera.position.fromArray(calibration.position);
+    // calibration.rotation is row-major but fromArray expects a column-major array, yielding the transposed matrix
+    var rotationInverse = new THREE.Matrix3().fromArray(calibration.rotation);
```

Comment:
Why not alter the `calibration.rotation` array, to be a 16-length array, and then create directly a `Matrix4` array ? Or at least you should declare the `rotationInverse` Matrix3 outside this function.

Reply:
A 4x4 rotation would have 7 constant entries (0s on the last lines and columns except a 1 on the lower right) which seems a bit redundant...
Ok for the memory optimization.

Reply:
Yeah I was kind of conflicted on this one, just move the `new` outside the method.